wandb/sdk/launch/utils.py

```diff
@@ -424,20 +424,28 @@ def _fetch_git_repo(dst_dir: str, uri: str, version: Optional[str]) -> None:
     repo = git.Repo.init(dst_dir)
     origin = repo.create_remote("origin", uri)
     origin.fetch()
-    if version is not None:
+
+    if version:
         try:
             repo.git.checkout(version)
         except git.exc.GitCommandError as e:
             raise ExecutionError(
                 "Unable to checkout version '%s' of git repo %s"
-                "- please ensure that the version exists in the repo. \n"
+                "- please ensure that the version exists in the repo. "
                 "Error: %s" % (version, uri, e)
             )
     else:
-        version = "master"  # change to "main" for current github practice?
+        # Check if main is in origin, else set branch to master
+        branches = [ref.name for ref in repo.references]
```

Comment:
if repo doesn't exist repo.references might error here. Is it possible to reach this point with a broken/nonexistent git repo?